### ISSUE TITLE
Fix S2583 FP: Assignment through tuple is not taken into consideration.

### ISSUE THREAD

**Issue by omarjuul** [SonarSource/sonar-dotnet#8043]
### Description

FP S2583 when overwriting variable via Tuple. Might have to do with migration to Symbolic Execution engine?

### Repro steps

``` csharp
private static void Test()
{
    bool testBool = false;
    string foo = "TRUE";
    (testBool, foo) = SomeWork(testBool, foo);
    if (testBool) // Noncompliant FP
    {
        Console.WriteLine(foo);
    }

    static (bool, string) SomeWork(bool dummy, string foo) => (true, $"Hello {foo}");
}
```
(obviously this code doesn't make much sense but it's a minimal repro of more complicated existing code that does make sense)

### Expected behavior

No trigger on rule S2583 expected.

### Actual behavior

Rule S2583 falsely thinks `testBool` is still false.

### Known workarounds

We could suppress the warning, or stay on version 9.7 of SonarAnalyzer (which is what we will do for now).

### Related information

* SonarAnalyzer version `9.10.0.77988` (False Positive not present in version `9.7`)

**Comment by mary-georgiou-sonarsource** [comment]
Hello again @omarjuul,

This is also indeed an FP.
I added it to our backlog to handle in anext hardening sprint.

**Comment by mary-georgiou-sonarsource** [comment]
Another report related to the tuples: https://community.sonarsource.com/t/switch-expression-s2589-false-postive-or-documentation-gap/99560

**Comment by mary-georgiou-sonarsource** [comment]
Closing this issue as duplicate of this: https://github.com/SonarSource/sonar-dotnet/issues/8027
